pow/src/main/java/tech/pegasys/teku/pow/fallback/FallbackAwareEth1Provider.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.pow.fallback;
+
+import java.math.BigInteger;
+import java.time.Duration;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.function.Function;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.web3j.protocol.core.methods.response.EthBlock.Block;
+import org.web3j.protocol.core.methods.response.EthCall;
+import tech.pegasys.teku.infrastructure.async.SafeFuture;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.pow.Eth1Provider;
+import tech.pegasys.teku.pow.fallback.strategy.Eth1ProviderSelector;
+
+public class FallbackAwareEth1Provider implements Eth1Provider {
+  private static final Logger LOG = LogManager.getLogger();
+  private final Eth1ProviderSelector eth1ProviderSelector;
+
+  public FallbackAwareEth1Provider(final Eth1ProviderSelector eth1ProviderSelector) {
+    this.eth1ProviderSelector = eth1ProviderSelector;
+  }
+
+  @Override
+  public SafeFuture<Optional<Block>> getEth1Block(final UInt64 blockNumber) {
+    return run(eth1Provider -> eth1Provider.getEth1Block(blockNumber));
+  }
+
+  @Override
+  public SafeFuture<Optional<Block>> getEth1BlockWithRetry(
+      final UInt64 blockNumber, final Duration retryDelay, final int maxRetries) {
+    return run(
+        eth1Provider -> eth1Provider.getEth1BlockWithRetry(blockNumber, retryDelay, maxRetries));
```

Comment:
These calls that automatically retry need to be handled a bit smarter - because `Web3jEth1Provider.getEth1BlockWithRetry` and the `getGuaranteedEth1Block` methods automatically retry, we wind up delegating the call to the first Eth1Provider which continuously retries even though its down and we never move on to the other providers.  I think we need to have these methods delegate to the plain `getEth1Block` but then handle the retry logic here - particularly for `getGuaranteedEth1Block` it just keeps retrying until it eventually gets the block, whereas `getEth1BlockWithRetry` retries a certain number of times then gives up.